File: Cpp solutions/1 two sum.cpp
# https://leetcode.com/problems/two-sum
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> indexes;
        map<int,int> numbersMap;
        int pairIndex = 0;
        int index = 0;
        while (pairIndex == 0 && index < nums.size()) {
            pairIndex = numbersMap[target - nums[index]];
            if (pairIndex == 0){
                numbersMap[nums[index]] = index + 1;
            }
            else{
                indexes.push_back(pairIndex-1);
                indexes.push_back(index);
            }
            index++;
        }
        return indexes;
    }
};

File: Python solutions/1 two sum.py
# https://leetcode.com/problems/two-sum
# TC: o(n) SC: o(n)
# Hashmap
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prev = {}
        for i in range(len(nums)):
            if target - nums[i] in prev:
                return [prev[target - nums[i]], i]
            prev[nums[i]] = i
        return [-1]

